utils/socket.h

```diff
@@ -1,6 +1,8 @@
 #ifndef UFTRACE_SOCKET_H
 #define UFTRACE_SOCKET_H
 
+#include "uftrace.h"
```

Comment:
But it also can make the dependency complicated.  For a simple type declaration for a pointer type, I'd recommend to have a forward declaration like instead.

```c
struct uftrace_msg;
```

Reply:
Right. Forward declaration is better in this  case.